test/tld.js

```diff
@@ -90,6 +90,31 @@ describe('tld.js', function () {
     });
   });
 
+  describe('isIp method', function () {
+    it('should return false on incorrect inputs', function () {
+      expect(tld.isIp('')).to.be(false);
+      expect(tld.isIp(null)).to.be(false);
+      expect(tld.isIp(undefined)).to.be(false);
+      expect(tld.isIp({})).to.be(false);
+    });
+
+    it('should return true on valid ip addresses', function () {
+      expect(tld.isIp('::1')).to.be(true);
+      expect(tld.isIp('2001:0db8:85a3:0000:0000:8a2e:0370:7334')).to.be(true);
+      expect(tld.isIp('192.168.0.1')).to.be(true);
```

Comment:
Would it make a difference if there is a `port` in the IP address?

Reply:
Not really. `isIp` will just check that some properties of the `hostname` that usually indicates an ip address are met. The `port` would be stripped during the parsing of the URL itself, before we check if the input is an `ip` so that should not be an issue in practice.